omnigibson/systems/micro_particle_system.py

```diff
@@ -672,13 +564,21 @@ def reset(cls):
         # Reset all internal variables
         cls.remove_all_particle_instancers()
 
-    @classmethod
-    def update_max_instancer_idn(cls):
+    @classproperty
+    def next_available_instancer_idn(cls):
         """
         Updates the max instancer identification number based on the current internal state
         """
-        cls.max_instancer_idn = -1 if len(cls.particle_instancers) == 0 else \
-            int(np.max([cls.particle_instancer_name_to_idn(name) for name in cls.particle_instancers.keys()]))
+        if cls.n_instancers == 0:
+            return cls.default_instancer_idn
+        else:
+            for idn in range(max(cls.instancer_idns) + 2):
+                if idn not in cls.instancer_idns:
+                    return idn
```

Comment:
i wonder if a set operation would be faster here:

`idn = np.min(list(set(range(max(cls.instancer_idns) + 2)) - set(cls.instancer_idns)))`

Reply:
I tested it. The speed is pretty comparable. I will just leave it be because it's slightly more interpretable

Reply:
sg!